config/database.go

```diff
@@ -234,33 +236,47 @@ func (ds *DatabaseStore) persist(cfg *model.Config) error {
 	}
 	defer func() {
 		// Rollback after Commit just returns sql.ErrTxDone.
-		if err := tx.Rollback(); err != nil && err != sql.ErrTxDone {
+		if err = tx.Rollback(); err != nil && err != sql.ErrTxDone {
```

Comment:
(Sorry GitHub won't let me comment earlier)

Can we move the transaction section to above the `tx.QueryRow` part? Just to avoid the check SHA part.